### PR TITLE
KOGITO-2417 ui-packages for building the react app

### PR DESCRIPTION
**Pull request by jstastny-cz**
Variant of #264

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -65,5 +71,6 @@
     <module>management-console</module>
     <module>explainability-service</module>
     <module>trusty</module>
+    <module>ui-packages</module>
```
Comment (cristianonicolai) [inline pom.xml]: technically it shouldnt make difference but can you add it before the management-console module please.

File: ui-packages/pom.xml [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    <sonar.inclusions>*/src/**</sonar.inclusions>
+    <sonar.test.inclusions>**/*test.ts,**/*test.tsx</sonar.test.inclusions>
+    <sonar.javascript.lcov.reportPaths>
+      ${path.to.packages}/common/coverage/lcov.info,
+      ${path.to.packages}/management-console/coverage/lcov.info,
+      ${path.to.packages}/task-console/coverage/lcov.info
+    </sonar.javascript.lcov.reportPaths>
+  </properties>
+  
+  <dependencyManagement>
```
Comment (cristianonicolai) [inline ui-packages/pom.xml]: @jstastny-cz do we need these deps here?

File: ui-packages/packages/common/index.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+export {
```
Comment (cristianonicolai) [inline ui-packages/packages/common/index.ts]: this is the only file that is showing as new instead of a move, is that just the github UI?

**Review by cristianonicolai — COMMENTED** [review]
@jstastny-cz looks better to me

File: ui-packages/packages/common/index.ts (hunk 3 above)
Comment (jstastny-cz) [reply ui-packages/packages/common/index.ts]: It is some git magic, I've tried to revert and use `git mv`, but the result was the same. I don't understand why this happens especially when the file is 1-1 the same.

File: pom.xml (hunk 1 above)
Comment (jstastny-cz) [reply pom.xml]: sure, no problem.

File: ui-packages/packages/common/index.ts (hunk 3 above)
Comment (jstastny-cz) [reply ui-packages/packages/common/index.ts]: Ah! It got reformatted heavily. I just moved it :-D . That's the crazy stuff to make our code pretty doing this!

File: ui-packages/pom.xml (hunk 2 above)
Comment (jstastny-cz) [reply ui-packages/pom.xml]: Nope, correct, bad copy&paste -> we don't use this as parent or use any deps directly. Will remove.

Comment (jstastny-cz) [reply ui-packages/pom.xml]: On second look the version for eirslett might be set there ... otherwise I'd expect the build to fail - will investigate.

Comment (jstastny-cz) [reply ui-packages/pom.xml]: Nope, it's managed in kogito-apps/pom.xml removing section.

**Review by cristianonicolai — APPROVED** [review]

**Comment by jstastny-cz** [comment]
Still one outstanding issue during build, investigating.
```
[ERROR] Install TypeScript in the project directory or use NODE_PATH env. variable to set TypeScript location, if it's located outside of project directory.
```

**Review by pefernan — APPROVED** [review]
+1 to merge... It will cause a lot of pain on my PR but we are here to play!

**Review by nmirasch — APPROVED** [review]
Really nice! Thanks Jan

**Pull request merged**